robocop/checkers/tags.py

```diff
@@ -356,3 +399,69 @@ def visit_Tags(self, node):  # noqa
                 lineno=tag.lineno,
                 col=tag.col_offset + 1,
             )
+
+
+class KeywordTagsChecker(VisitorChecker):
+    """Checker for keyword tags."""
+
+    reports = (
+        "could-be-keyword-tags",
+        "tag-already-set-in-keyword-tags",
+    )
+
+    def __init__(self):
+        self.keywords_tags = []
+        self.keyword_tags_node = None
+        self.in_keywords = False
+        super().__init__()
+
+    def visit_File(self, node):  # noqa
+        self.keywords_tags = []
+        self.keyword_tags = set()
+        self.keywords_count = 0
+        self.keyword_tags_node = None
+        super().visit_File(node)
+        if not self.keywords_tags:
+            return
+        if len(self.keywords_tags) != self.keywords_count:
+            return
+        if self.keywords_count < 2:
+            return
+        common_keyword_tags = set.intersection(*[set(tags) for tags in self.keywords_tags])
+        common_keyword_tags = common_keyword_tags - self.keyword_tags
+        if common_keyword_tags:
+            report_node = node if self.keyword_tags_node is None else self.keyword_tags_node
+            self.report(
+                "could-be-keyword-tags",
+                tags=", ".join(common_keyword_tags),
+                node=report_node,
+            )
+
+    def visit_Keyword(self, node):  # noqa
+        self.keywords_count += 1
+        self.generic_visit(node)
+
+    def visit_KeywordTags(self, node):  # noqa
+        self.keyword_tags = {token.value for token in node.data_tokens[1:]}
+        self.keyword_tags_node = node
+
+    def visit_KeywordSection(self, node):  # noqa
+        self.in_keywords = True
+        self.generic_visit(node)
+        self.in_keywords = False
+
+    def visit_Tags(self, node):  # noqa
+        if self.in_keywords:
+            self.keywords_tags.append([tag.value for tag in node.data_tokens[1:]])
```

Comment:
Keywords_tags is sometimes set and sometimes list, what type it should be?

Reply:
I think you might be confused because there are `self.keyword_tags` (for storing all tags set by `Keyword Tags` setting) and `self.keywords_tags` (for storing all tags set in keywords' `[Tags]` setting). I renamed the latter to `self.tags_in_keywords` to avoid confusion.

Also, the code is heavily copied from the corresponding test-related rules for tags.

Reply:
Ah, you're right - didn't notice the extra s. Thanks for the rename